src/backend/distributed/commands/create_citus_local_table.c

```diff
@@ -0,0 +1,911 @@
+ * owned sequence, it grants ownership to the same named column of the relation
+ * with targetRelationId.
+ */
+static void
+DropAndMoveDefaultSequenceOwnerships(Oid sourceRelationId, Oid targetRelationId)
+{
+	List *columnNameList = NIL;
+	List *ownedSequenceIdList = NIL;
+	ExtractColumnsOwningSequences(sourceRelationId, &columnNameList,
+								  &ownedSequenceIdList);
+
+	ListCell *columnNameCell = NULL;
+	ListCell *ownedSequenceIdCell = NULL;
+	forboth(columnNameCell, columnNameList, ownedSequenceIdCell, ownedSequenceIdList)
+	{
+		char *columnName = (char *) lfirst(columnNameCell);
+		Oid ownedSequenceId = lfirst_oid(ownedSequenceIdCell);
+
+		DropDefaultColumnDefinition(sourceRelationId, columnName);
+
+		/* column might not own a sequence */
+		if (OidIsValid(ownedSequenceId))
+		{
+			TransferSequenceOwnership(ownedSequenceId, targetRelationId, columnName);
+		}
+	}
+}
+
+
+/*
+ * ExtractColumnsOwningSequences finds each column of relation with relationId
+ * defaulting to an owned sequence. Then, appends the column name and id of the
+ * owned sequence -that the column defaults- to the lists passed as NIL initially.
+ */
+static void
+ExtractColumnsOwningSequences(Oid relationId, List **columnNameList,
+							  List **ownedSequenceIdList)
+{
+	Assert(*columnNameList == NIL && *ownedSequenceIdList == NIL);
+
+	Relation relation = relation_open(relationId, AccessShareLock);
+	TupleDesc tupleDescriptor = RelationGetDescr(relation);
+
+	for (int attributeIndex = 0; attributeIndex < tupleDescriptor->natts;
+		 attributeIndex++)
+	{
+		Form_pg_attribute attributeForm = TupleDescAttr(tupleDescriptor, attributeIndex);
+		if (attributeForm->attisdropped || !attributeForm->atthasdef)
+		{
+			/*
+			 * If this column has already been dropped or it has no DEFAULT
+			 * definition, skip it.
+			 */
+			continue;
+		}
+
+		char *columnName = NameStr(attributeForm->attname);
+		*columnNameList = lappend(*columnNameList, columnName);
+
+		List *columnOwnedSequences =
+			GetSequencesOwnedByColumn(relationId, attributeIndex + 1);
+
+		Oid ownedSequenceId = InvalidOid;
+		if (list_length(columnOwnedSequences) != 0)
+		{
+			/*
+			 * A column might only own one sequence. We intentionally use
+			 * getOwnedSequences and pick initial oid from the list instead
+			 * of using getOwnedSequence as it errors out if column does not
+			 * have any sequences.
+			 */
+			Assert(list_length(columnOwnedSequences) == 1);
+			ownedSequenceId = linitial_oid(columnOwnedSequences);
+		}
+
+		*ownedSequenceIdList = lappend_oid(*ownedSequenceIdList, ownedSequenceId);
+	}
+
+	relation_close(relation, NoLock);
+}
+
+
+/*
+ * GetSequencesOwnedByColumn returns a list of OIDs for all the sequences that
+ * the column with attributeNumber of relation with relationId implicitly depends
+ * (DEPENDENCY_AUTO or DEPENDENCY_INTERNAL).
+ */
+static List *
+GetSequencesOwnedByColumn(Oid relationId, AttrNumber attributeNumber)
+{
+#if PG_VERSION_NUM >= PG_VERSION_13
```

Comment:
The #if directives in the code don't look that nice, so when possible, it is better to handle that with some compat version. For example here we could define a macro such as `getOwnedSequencesCompat ` in https://github.com/citusdata/citus/blob/ba208eae4d19e85cdfa22b527c90ace406e70bfd/src/include/distributed/version_compat.h#L31 and also in https://github.com/citusdata/citus/blob/ba208eae4d19e85cdfa22b527c90ace406e70bfd/src/include/distributed/version_compat.h#L59. 

The macros could be:

```c
#if PG_VERSION_NUM >= PG_VERSION_13
getOwnedSequencesCompat(a,b) getOwnedSequences_internal(a, b, 0);
....
#else 
getOwnedSequencesCompat(a,b)  getOwnedSequences(a, b);

```

That way here in the code we wouldn't have these details and here we can just do:

```c
return getOwnedSequencesCompat(relationId, attributeNumber);
```

If the comments are really necessary, we can move them to the definition of macros however we already have a documentation of the method `GetSequencesOwnedByColumn`

Reply:
see https://github.com/citusdata/citus/blob/cba4d072bbcdd1566272487c058052d7c2ee933c/src/include/distributed/version_compat.h#L37-L38 & https://github.com/citusdata/citus/blob/cba4d072bbcdd1566272487c058052d7c2ee933c/src/include/distributed/version_compat.h#L52-L53